engine/executor/dm/worker.go

```diff
@@ -231,7 +233,12 @@ func (w *dmWorker) tryUpdateStatus(ctx context.Context) error {
 			return nil
 		}
 	}
-	return w.Exit(ctx, status, nil)
+
+	if err := w.Exit(ctx, framework.ExitReasonFinished, nil, status.ExtBytes); err != nil {
+		return err
+	}
+
+	return derror.ErrWorkerFinish.FastGenByArgs()
```

Comment:
Why we return a error here?

Reply:
The return sematic of `Exit` has changed after this pr.
Before:
`Exit` always return an error
After:
`Exit` return an error when it encounters something wrong. 

So to keep the sematic of `tryUpdateStatus`, I return an `ErrWorkerFinish` here.